File: sample_apps/javaV2/src/main/java/com/amazonaws/services/timestream/BasicExample.java
package com.amazonaws.services.timestream;

import java.io.IOException;

import software.amazon.awssdk.services.timestreamquery.TimestreamQueryClient;
import software.amazon.awssdk.services.timestreamwrite.TimestreamWriteClient;

import com.amazonaws.services.timestream.utils.WriteUtil;

import static com.amazonaws.services.timestream.utils.Constants.DATABASE_NAME;
import static com.amazonaws.services.timestream.utils.Constants.TABLE_NAME;

public class BasicExample {
    private final InputArguments inputArguments;
    private final TimestreamWriteClient writeClient;
    private final TimestreamQueryClient queryClient;
    private final CrudAndSimpleIngestionExample crudAndSimpleIngestionExample;
    private final CsvIngestionExample csvIngestionExample;
    private final QueryExample queryExample;
    private final WriteUtil writeUtil;

    public BasicExample(InputArguments inputArguments, final TimestreamWriteClient writeClient, final TimestreamQueryClient queryClient) {
        this.inputArguments = inputArguments;
        this.writeClient = writeClient;
        this.queryClient = queryClient;
        this.writeUtil = new WriteUtil(writeClient);
        this.crudAndSimpleIngestionExample = new CrudAndSimpleIngestionExample(writeClient, writeUtil);
        this.csvIngestionExample = new CsvIngestionExample(writeClient);
        this.queryExample = new QueryExample(queryClient);
    }

    public void run() throws IOException {
        this.crudAndSimpleIngestionExample.run(inputArguments);
        if (inputArguments.getInputFile() != null) {
            // Bulk record ingestion for bootstrapping a table with fresh data
            csvIngestionExample.bulkWriteRecords(inputArguments.getInputFile());
        }
        try {
            // Query samples
            queryExample.runAllQueries();

            // Try cancelling a query
            queryExample.cancelQuery();

            // Run a query with Multiple pages
            queryExample.runQueryWithMultiplePages(20000);
        } catch (Exception e) {
            // Some queries might fail with 500 if the result of a sequence function has more than 10000 entries
            e.printStackTrace();
        } finally {
            if (!inputArguments.isSkipDeletion()) {
                writeUtil.deleteTable(DATABASE_NAME, TABLE_NAME);
                writeUtil.deleteDatabase(DATABASE_NAME);
            }
        }
    }
}

File: sample_apps/python/live_analytics/basic_example/BasicExample.py
from Constant import DATABASE_NAME, TABLE_NAME
from utils.WriteUtil import WriteUtil
from CrudAndSimpleIngestionExample import CrudAndSimpleIngestionExample
from CsvIngestionExample import CsvIngestionExample
from QueryExample import QueryExample
class BasicExample:
    def __init__(self, write_client, query_client, kmsId, csv_file_path, skip_deletion):
        self.kmsId = kmsId
        self.csv_file_path = csv_file_path
        self.skip_deletion = skip_deletion
        self.crud_and_simple_ingestion_example = CrudAndSimpleIngestionExample(write_client)
        self.csv_ingestion_example = CsvIngestionExample(write_client)
        self.query_example = QueryExample(query_client)
        self.write_util = WriteUtil(write_client)

    def run(self):
        try:
            self.crud_and_simple_ingestion_example.run(self.kmsId)
            if self.csv_file_path != None:
                self.csv_ingestion_example.bulk_write_records(self.csv_file_path)

            self.query_example.run_all_queries()

            # Try cancelling a query
            self.query_example.cancel_query()

            # Try a query with multiple pages
            self.query_example.run_query_with_multiple_pages(20000)
        finally:
            if not self.skip_deletion:
                self.write_util.delete_table(DATABASE_NAME, TABLE_NAME)
                self.write_util.delete_database(DATABASE_NAME)

